exporter/datadogexporter/internal/logs/sender.go

```diff
@@ -64,30 +65,62 @@ func (s *Sender) SubmitLogs(ctx context.Context, payload []datadogV2.HTTPLogItem
 		s.logger.Debug("Submitting logs", zap.Any("payload", payload))
 	}
 
-	// enable sending gzip
-	// Get a fresh opts for each request to avoid duplicating ddtags
-	opts := *datadogV2.NewSubmitLogOptionalParameters().WithContentEncoding(datadogV2.CONTENTENCODING_GZIP)
+	var prevDdtags string
+	var logItemBatch []datadogV2.HTTPLogItem
+
+	if len(payload) > 0 && payload[0].HasDdtags() {
+		prevDdtags = *payload[0].Ddtags
+	}
 
 	// Correctly sets apiSubmitLogRequest ddtags field based on tags from translator Transform method
-	if payload[0].HasDdtags() {
-		tags := datadog.PtrString(payload[0].GetDdtags())
-		if opts.Ddtags != nil {
-			tags = datadog.PtrString(fmt.Sprint(*opts.Ddtags, ",", payload[0].GetDdtags()))
+	for i, logItem := range payload {
```

Comment:
I would either rename `payload` to `logItems` (which makes complete sense). Or rename `logItem` to `p`.